jkube-kit/config/image/src/test/java/org/eclipse/jkube/kit/config/image/ImageNameDistributionReferenceTest.java

```diff
@@ -0,0 +1,108 @@
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+/**
+ * Tests ported from <a href="https://github.com/distribution/reference/blob/main/reference_test.go">Distribution Reference</a>
+ */
+class ImageNameDistributionReferenceTest {
+
+  @ParameterizedTest
+  @ValueSource(strings = {
+      "test.com/foo",
+      "test:8080/foo",
+      "docker.io/library/foo",
+      "test_com",
+      "test.com:tag",
+      "test.com:5000",
+      "test.com/repo:tag",
+      "test:5000/repo",
+      "test:5000/repo:tag",
+      "lowercase:Uppercase",
+      "test:5000/repo@sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+      "test:5000/repo:tag@sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff",
+      "sub-dom1.foo.com/bar/baz/quux",
+      "sub-dom1.foo.com/bar/baz/quux:some-long-tag",
+      "b.gcr.io/test.example.com/my-app:test.example.com",
+      "xn--n3h.com/myimage:xn--n3h.com",
+      //"xn--7o8h.com/myimage:xn--7o8h.com@sha512:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff", // Digest SHA 512
+      "foo_bar.com:8080",
+      "foo/foo_bar.com:8080",
+      "192.168.1.1",
+      "192.168.1.1:tag",
+      "192.168.1.1:5000",
+      "192.168.1.1/repo",
+      "192.168.1.1:5000/repo",
+      "192.168.1.1:5000/repo:5050",
+      "a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a/a:tag-puts-this-over-max",
+      "docker/docker",
+      "library/debian",
+      "debian",
+      "localhost/library/debian",
+      "localhost/debian",
+      "docker.io/docker/docker",
+      "docker.io/library/debian",
+      "docker.io/debian",
+      "index.docker.io/docker/docker",
+      "index.docker.io/library/debian",
+      "index.docker.io/debian",
+      "127.0.0.1:5000/docker/docker",
+      "127.0.0.1:5000/library/debian",
+      "127.0.0.1:5000/debian",
+      "192.168.0.1",
+      "192.168.0.1:80",
+      "192.168.0.1:8/debian",
+      "192.168.0.2:25000/debian",
+      "docker.io/1a3f5e7d9c1b3a5f7e9d1c3b5a7f9e1d3c5b7a9f1e3d5d7c9b1a3f5e7d9c1b3a",
+      //"[2001:db8::1]/repo", // Ipv6
+      //"[2001:db8:1:2:3:4:5:6]/repo:tag", // Ipv6
+      //"[2001:db8::1]:5000/repo", // Ipv6
+      //"[2001:db8::1]:5000/repo:tag", // Ipv6
+      //"[2001:db8::1]:5000/repo@sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff", // Ipv6
+      //"[2001:db8::1]:5000/repo:tag@sha256:ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff", // Ipv6
+      //"[2001:db8::]:5000/repo", // Ipv6
+      //"[::1]:5000/repo", // Ipv6
```

Comment:
Does `// IPv6` imply  what's stated later `Invalid Ipv6 addresses - these are failing because we only support IPv4 at the moment`
In my previous comment I was requesting the reason why the test is disabled. IPv5 is not a reason. A reason might be that IPv6 is not supported on upstream Docker as a valid image name, or whatever the **justifiable** reason is why we're no supporting IPv6.

Reply:
okay, let me add more detail to this comment. These are valid Ipv6 addresses that ImageName fails to parse. I checked docker source and I saw that they now do have support for Ipv6 addresses but it's not available on the latest version v24.0.8. Maybe it would be available in upcoming releases.